revorder_test.go

```diff
@@ -166,18 +135,9 @@ func TestRevOrderPanicEvict(t *testing.T) {
 
 	select {
 	case r := <-recoverChan:
-		if r == nil {
-			t.Errorf("nil panic/recover, expected non-nil recover() value")
-		} else {
-			pat1 := "Entity abc evicted"
-			if !strings.Contains(r.(string), pat1) {
-				t.Errorf("panic msg doesn't contain '%s': %s", pat1, r)
-			}
-			pat2 := "revisions > 2. Received revisions: 4"
-			if !strings.Contains(r.(string), pat2) {
-				t.Errorf("panic msg doesn't contain '%s': %s", pat2, r)
-			}
-		}
+		require.NotNil(t, r)
+		assert.Contains(t, r.(string), "Entity abc evicted")
+		assert.Contains(t, r.(string), "revisions > 2. Received revisions: 4")
 	case <-time.After(1 * time.Second):
```

Comment:
I think you can use `assert.Eventually` for this block.

Reply:
That's cool, I hadn't seen assert.Eventually before. I tried rewriting it with eventually as follows: 
```
	var r interface{}
	require.Eventually(t, func() bool {
		select {
		case r = <-recoverChan:
			return true
		default:
			return false
		}
	}, 1*time.Second, 10*time.Millisecond)
	require.NotNil(t, r)
	assert.Contains(t, r.(string), "Entity abc evicted")
	assert.Contains(t, r.(string), "revisions > 2. Received revisions: 4")
```

But I think that's less readable than the original. I can see where this could be useful in other cases though, glad to know it.